src/net/sourceforge/kolmafia/textui/command/AbstractModeCommand.java

```diff
@@ -0,0 +1,32 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import java.util.Set;
+
+public abstract class AbstractModeCommand extends AbstractCommand {
+  /**
+   * Basic side-effect-less command validation
+   *
+   * @param command Command string
+   * @param parameters Parameters as a string
+   * @return Whether command is valid
+   */
+  public boolean validate(final String command, final String parameters) {
+    return true;
```

Comment:
if this is never called at all and always overridden, `AbstractModeCommand` might be better as an interface.